finality-aleph/src/testing/data_store.rs

```diff
@@ -294,7 +294,12 @@ async fn branch_not_within_session_boundaries_does_not_go_through() {
         .assert_no_message_out("Data Store let through a message not within session_boundaries")
         .await;
 
-    test_handler.finalize_block(&blocks[session_end + MAX_DATA_BRANCH_LEN].hash());
+    test_handler.finalize_block(
+        &blocks[session_end + MAX_DATA_BRANCH_LEN]
+            .header()
+            .id()
+            .hash(),
```

Comment:
Kinda related to the above comment – perhaps the interface here should be accepting ids rather than hashes, to be more conformant with our id-first approach in the main code. But this might be unnecessary work, so ehh~~, I encourage you to silently resolve this comment and pretend you didn't see it, unless you are very much in the mood for changing this.

Reply:
[I tried :P] After a consideration, I think this'd be better done after settling block production and finalization abstractions. The whole `ClientChainBuilder` will have to be refactored, and this `TestHandler` too, as a consequence.